File: app/src/main/java/com/walmartlabs/task/model/WalmartApplication.java
package com.walmartlabs.task.model;

import android.app.Application;

public class WalmartApplication extends Application {

    private static WalmartApplication mInstance;

    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;
    }
    public static synchronized WalmartApplication getInstance() {
        return mInstance;
    }

    public void setConnectivityListener(ConnectivityReceiver.ConnectivityReceiverListener listener) {
        ConnectivityReceiver.connectivityReceiverListener = listener;
    }
}

File: app/src/main/java/com/walmartlabs/task/model/WalmartApplication.kt
package com.walmartlabs.task.model

import android.app.Application

/**
 * Created by Bipul on 25-08-2019.
 */
class WalmartApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    fun setConnectivityListener(listener: ConnectivityReceiver.ConnectivityReceiverListener) {
        ConnectivityReceiver.connectivityReceiverListener = listener
    }

    companion object {

        @get:Synchronized
        var instance: WalmartApplication? = null
            private set
    }
}